ow_power_system/src/power_system_node.cpp

```diff
@@ -14,91 +14,131 @@
-  ros::Publisher RUL_pub = nh.advertise<std_msgs::Int16>("remaining_useful_life",1000);
-  //Construct our Battery Temperature (TempBat) publisher
-  ros::Publisher TempBat_pub = nh.advertise<std_msgs::Float64>("battery_temperature",1000);
-  
-  //Load power values csv
-  string csv_path;
-  string default_csv = "/data/data_const_load.csv";
-  bool csv_path_param_exist = nh.param("power_draw_csv_path", csv_path,
-    ros::package::getPath("ow_power_system") + default_csv);
-
-  if (!csv_path_param_exist)
+using namespace chrono;
+
+double update_count = 0.00; // Timestep initialization
+double battery_lifetime = 2738.0; // Estimate of battery lifetime (seconds)
+Datum<double> init_power = 0.00; // Power initialization
+Datum<double> init_temp = 20.00; // Temperature initialization
+Datum<double> init_voltage = 4.10; // Voltage initialization
+auto init_time = system_clock::now(); // Timestep initialization
+
+class SubscribeAndPublish
+{
+public:
+  SubscribeAndPublish()
   {
-    ROS_WARN_NAMED("power_system_node", "power_draw_csv_path param was not set! Using default value: data_const_load.csv");
-  }
-
-  ROS_INFO_STREAM_NAMED("power_system_node", "power_draw_csv_path is set to: " << csv_path);
+    //Construct our State of Charge (SOC) publisher
+    SOC_pub = nh.advertise<std_msgs::Float64>("power_system_node/state_of_charge",1);
+    //Construct our Remaining Useful Life (RUL) publisher (Seconds)
+    RUL_pub = nh.advertise<std_msgs::Int16>("power_system_node/remaining_useful_life",1);
+    //Construct our Battery Temperature (TempBat) publisher
+    TempBat_pub = nh.advertise<std_msgs::Float64>("power_system_node/battery_temperature",1);
 
-  // Read battery data from a file.
-  auto power_data = read_file(csv_path);
+    //Construct our Average Mechanical Power subsriber (Watts)
+    MechPower_sub = nh.subscribe("/mechanical_power/average", 1, &SubscribeAndPublish::powerCallback, this);
 
-  // Create a configuration from a file
-  string config_path = ros::package::getPath("ow_power_system")+"/config/example.cfg";
-  ConfigMap config(config_path);
-  // Contruct a new prognoser using the prognoser factory. The prognoser
-  // will automatically construct an appropriate model, observer and predictor
-  // based on the values specified in the config.
-  auto prognoser = PrognoserFactory::instance().Create("ModelBasedPrognoser", config);
-
-  // Retrieve our publication rate expressed in Hz
-  double power_update_rate = 1;
-  double power_update_rate_override;  // allow the user to override it
-  bool update_rate_param_exist = nh.param("power_update_rate", power_update_rate_override, 0.1);
-
-  if (update_rate_param_exist)
+  }
+  void powerCallback(const std_msgs::Float64::ConstPtr& msg)
   {
-    ROS_WARN_NAMED("power_system_node", "Overriding the default power_update_rate!");
-    // Validated the parameter
-    if (power_update_rate == 0)
+    using namespace std::chrono;
```

Comment:
duplicate statement .. you have already defined this clause early in file
```suggestion
```

Reply:
Resolved in latest commit.